src/main/java/com/terraformersmc/campanion/network/C2SRotateHeldItem.java

```diff
@@ -46,10 +46,11 @@ public static void onPacket(MinecraftServer server, ServerPlayerEntity player, S
 					tag.put("BlockState", NbtHelper.fromBlockState(rotatedState));
 
 					if (!data.isEmpty()) {
-						BlockEntity be = BlockEntity.createFromTag(state, data);
+						BlockEntity be = BlockEntity.createFromNbt(rotatedPos, state, data);
 						if (be != null) {
-							be.applyRotation(BlockRotation.CLOCKWISE_90);
-							tag.put("BlockEntityData", be.toTag(new CompoundTag()));
+							// TODO - Can't find equivilent
```

Comment:
What does the trident do? That's what this was based on before I think

Reply:
The only use of this in 1.16 I could find was in Structure.java, around line 366. In 1.17 the references to applyMirror() and applyRotation() to the block entity have been removed and replaced with nothing that I can see. My best guess is that the readNbt() on the rotated block state means that this is no longer necessary, but the only way to prove this would be to test it. 
I'm assuming that this is responsible for making sure the block entity for the tent is at the correct rotation for the server as well as the client based on reading it? I'm not very familiar with minecraft networking

Reply:
I thought it was used to make the spear look right in the hand similar to the trident, but it's possible this is used for the tent